test/edu-benefits/1995/config/servicePeriods.unit.spec.jsx

```diff
@@ -0,0 +1,104 @@
+  const definitions = formConfig.defaultDefinitions;
+  it('should render', () => {
+    const form = ReactTestUtils.renderIntoDocument(
+      <DefinitionTester
+          schema={schema}
+          data={{}}
+          uiSchema={uiSchema}
+          definitions={definitions}/>
+    );
+
+    expect(ReactTestUtils.scryRenderedDOMComponentsWithTag(form, 'input'))
+      .to.not.be.empty;
+  });
+
+  it('should render service fields', () => {
+    const form = ReactTestUtils.renderIntoDocument(
+      <DefinitionTester
+          schema={schema}
+          data={{}}
+          uiSchema={uiSchema}
+          definitions={definitions}/>
+    );
+    const formDOM = findDOMNode(form);
+    const newServiceInput = Array.from(formDOM.querySelectorAll('input[type=radio]'))[0];
+
+    // just yes/no
+    expect(ReactTestUtils.scryRenderedDOMComponentsWithTag(form, 'input').length).to.equal(2);
+    expect(ReactTestUtils.scryRenderedDOMComponentsWithTag(form, 'select')).to.be.empty;
+
+    ReactTestUtils.Simulate.change(newServiceInput, {
+      target: {
+        value: 'Y'
+      }
+    });
+
+    // yes/no and service period fields (text box and two dates)
+    expect(ReactTestUtils.scryRenderedDOMComponentsWithTag(form, 'input').length).to.equal(5);
+    expect(ReactTestUtils.scryRenderedDOMComponentsWithTag(form, 'select').length).to.equal(4);
+  });
+
+  it('should render service period view', () => {
+    const form = ReactTestUtils.renderIntoDocument(
+      <DefinitionTester
+          schema={schema}
+          data={{
+            'view:newService': true,
+            toursOfDuty: [
+              {
+                serviceBranch: 'Army',
+                dateRange: {
+                  to: '2010-01-01',
+                  from: '2008-01-03'
+                }
+              },
+              {
+                dateRange: {}
+              }
+            ]
+          }}
+          uiSchema={uiSchema}
+          definitions={definitions}/>
+    );
+    const formDOM = findDOMNode(form);
+    const firstPeriod = Array.from(formDOM.querySelectorAll('.va-growable-background'))[0];
+
+    expect(firstPeriod.textContent).to.contain('Army');
+    expect(firstPeriod.textContent).to.contain('01/03/2008 — 01/01/2010');
+    expect(firstPeriod.querySelector('button').textContent).to.equal('Edit');
+  });
+  it('should have no required inputs', () => {
+    const onSubmit = sinon.spy();
+    const form = ReactTestUtils.renderIntoDocument(
+      <DefinitionTester
+          schema={schema}
+          onSubmit={onSubmit}
+          data={{}}
+          uiSchema={uiSchema}
+          definitions={definitions}/>
+    );
+    const formDOM = findDOMNode(form);
+    ReactTestUtils.findRenderedComponentWithType(form, Form).onSubmit({
+      preventDefault: f => f
+    });
+    expect(Array.from(formDOM.querySelectorAll('.usa-input-error'))).to.be.empty;
+
+    ReactTestUtils.findRenderedComponentWithType(form, Form).onSubmit({
```

Comment:
Could extract the `Form` finding out since we do it multiple times.

```js
const formElement = ReactTestUtils.findRenderedComponentWithType(form, Form);
formElement.onSubmit({ preventDefault: f => f });
```